imgix/urlbuilder.py

```diff
@@ -49,26 +49,37 @@ class UrlBuilder(object):
     """
     def __init__(
             self,
-            domains,
+            domains=None,
             use_https=True,
             sign_key=None,
             shard_strategy=SHARD_STRATEGY_CRC,
             sign_with_library_version=None,
-            include_library_param=True):
+            include_library_param=True,
+            domain=None):
 
         if sign_with_library_version is not None:
             warnings.warn('`sign_with_library_version` has been deprecated ' +
                           'and will be removed in the next major version. ' +
                           'Use `include_library_param` instead.',
                           DeprecationWarning, stacklevel=2)
 
-        if isinstance(domains, (list, tuple)) and (len(domains) > 1):
-            warnings.warn('Domain sharding has been deprecated and will ' +
-                          'be removed in the next major version. ',
-                          DeprecationWarning, stacklevel=2)
-
-        if not isinstance(domains, (list, tuple)):
-            domains = [domains]
+        if isinstance(domains, (list, tuple)):
+            if (len(domains) > 1):
+                warnings.warn('Domain sharding has been deprecated and will ' +
+                              'be removed in the next major version.\nAs a ' +
+                              'result, the \'domains\' argument will be ' +
+                              'deprecated in favor of \'domain\' instead.',
+                              DeprecationWarning, stacklevel=2)
+            elif (len(domains) == 0):
+                raise ValueError('Domains cannot take an empty array')
+        else:
+            if isinstance(domains, str):
+                domains = [domains]
```

Comment:
Do you think it would make sense to add the deprecation warning here too?

Reply:
I would say no, since in this scenario only a (single) string domain is being passed in - which is the more typical use case. By definition, users would only be making use of the domain sharding functionality if they passed in multiple domains to cycle through.

Reply:
Okay, sounds good!